applications/product/groovyScripts/facility/facility/CountFacilityInventoryByProduct.groovy

```diff
@@ -55,165 +57,168 @@ rows = [] as ArrayList
     if (monthsInPastLimitStr) {
         try {
             monthsInPastLimit = Integer.parseInt(monthsInPastLimitStr)
             cal = UtilDateTime.toCalendar(null)
             cal.add(Calendar.MONTH, 0 - monthsInPastLimit)
             checkTime = UtilDateTime.toTimestamp(cal.getTime())
-            searchParameterString += "&monthsInPastLimit=" + monthsInPastLimitStr
+            searchParameterString += '&monthsInPastLimit=' + monthsInPastLimitStr
         } catch (Exception e) {
-            logError(e, "Caught an exception : " + e.toString())
-            request.setAttribute("_ERROR_MESSAGE", "An exception occured please check the log")
+            logError(e, 'Caught an exception : ' + e)
+            request.setAttribute('_ERROR_MESSAGE', 'An exception occured please check the log')
         }
     }
 
     if (checkTime) {
-
         // Construct a dynamic view entity to search against for sales usage quantities
-        salesUsageViewEntity = new DynamicViewEntity()
-        salesUsageViewEntity.addMemberEntity("OI", "OrderItem")
-        salesUsageViewEntity.addMemberEntity("OH", "OrderHeader")
-        salesUsageViewEntity.addMemberEntity("ItIss", "ItemIssuance")
-        salesUsageViewEntity.addMemberEntity("InvIt", "InventoryItem")
-        salesUsageViewEntity.addViewLink("OI", "OH", false, ModelKeyMap.makeKeyMapList("orderId"))
-        salesUsageViewEntity.addViewLink("OI", "ItIss", false, ModelKeyMap.makeKeyMapList("orderId", "orderId", "orderItemSeqId", "orderItemSeqId"))
-        salesUsageViewEntity.addViewLink("ItIss", "InvIt", false, ModelKeyMap.makeKeyMapList("inventoryItemId"))
-        salesUsageViewEntity.addAlias("OI", "productId")
-        salesUsageViewEntity.addAlias("OH", "statusId")
-        salesUsageViewEntity.addAlias("OH", "orderTypeId")
-        salesUsageViewEntity.addAlias("OH", "orderDate")
-        salesUsageViewEntity.addAlias("ItIss", "inventoryItemId")
-        salesUsageViewEntity.addAlias("ItIss", "quantity")
-        salesUsageViewEntity.addAlias("InvIt", "facilityId")
+        salesUsageViewEntity = new DynamicViewEntity().with {
+            addMemberEntity('OI', 'OrderItem')
+            addMemberEntity('OH', 'OrderHeader')
+            addMemberEntity('ItIss', 'ItemIssuance')
+            addMemberEntity('InvIt', 'InventoryItem')
+            addViewLink('OI', 'OH', false, ModelKeyMap.makeKeyMapList('orderId'))
+            addViewLink('OI', 'ItIss', false, ModelKeyMap.makeKeyMapList('orderId', 'orderId', 'orderItemSeqId', 'orderItemSeqId'))
+            addViewLink('ItIss', 'InvIt', false, ModelKeyMap.makeKeyMapList('inventoryItemId'))
+            addAlias('OI', 'productId')
+            addAlias('OH', 'statusId')
+            addAlias('OH', 'orderTypeId')
+            addAlias('OH', 'orderDate')
+            addAlias('ItIss', 'inventoryItemId')
+            addAlias('ItIss', 'quantity')
+            addAlias('InvIt', 'facilityId')
+        }
```

Comment:
Need to return `it` for assignment to `salesUsageViewEntity`

Reply:
Done Thanks !